### PR TITLE
add in api calls to allow async sends. Fixes #615

### LINKED ISSUE #615: Send API dangerous and timeout does not mean a payment failed.
Seen examples of payments hanging for 4.5 hours before failing:

> 05:18:39.879 INFO  akka://default/user/$j/payment-initiator/$4 f.a.eclair.payment.PaymentLifecycle - route found: attempt=2/5 route=02ad6fb8d693dc1e4569bcedefadf5f72a931ae027dc0f0c544b34c1c6f3b9a02b-02d539692f55b6d61847dc432bb78166bf52318e80e6b33733bb75f76645656e27 channels=7f96a0003370000->803ed0006250001

> 09:54:44.878 INFO  akka://default/user/$j/payment-initiator/$4 f.a.eclair.payment.PaymentLifecycle - received an error message from local, trying to use a different channel (failure=channel is unavailable (offline or closing))

So do we need another call to poll the API to check if timed out payments have now failed or succeeded?

And actually in the lightning spec once we try to send a payment we can only be sure it has failed once the htlc is rolled back in our commit. So staying open for hours/days is a possibility that could be common?

### PR DESCRIPTION
**Pull request by n1bor**
Adds 2 API calls sendasync - this is identical to send but returns id instantly.
and checksentpayment - this takes the id as a parameter and return a result when there is one. If no results instantly returns nothing.

### CODE REVIEW

File: eclair-core/src/main/scala/fr/acinq/eclair/api/Service.scala [hunk 1]
```diff
@@ -276,33 +279,42 @@ trait Service extends Logging {
...
-                        }
+                      // check SENT payments
+                      case "checksentpayment" => req.params match {
+                        case JString(id) :: Nil => completeRpcFuture(req.id, (paymentInitiator ? SendPaymentId(id.toLong)).mapTo[CheckSendPaymentResult])
+                        case _ => reject(UnknownParamsRejection(req.id, "[sentPaymentId]"))
+                      }
 
-                        // method name was not found
-                        case _ => reject(UnknownMethodRejection(req.id))
+                      // check received payments
```
Comment (n1bor) [inline eclair-core/src/main/scala/fr/acinq/eclair/api/Service.scala]: Just fixed some indentation here. No real change to checkpayment.

**Comment by n1bor** [comment]
Any chance of getting this merged? Think I can move back onto trunk then...

**Comment by n1bor** [comment]
ping...

**Comment by sstone** [comment]
Sorry for not replying earlier. We've been looking at  ways to improve our API for some time, and I'd rather keep the issue open and not merge short-term solutions that have an impact on our internal design. The problem here is that when you send a payment you have to wait for an answer. Usually it's very quick, but it can take a long time when payments fail or are timed out. You should be able to achieve what you want with a call to our API and #694 (which will tell you when a payment has been sent and has a limited and more contained impact on our code base).

**Comment by n1bor** [comment]
Thanks - the sentinfo api call does exactly what I need.
So can send a payment and if it times out can just call this repeatedly to check if payed or failed.
Means I can make payments transactional.

**Pull request closed without merging**